djautotask/models.py

```diff
@@ -52,13 +71,103 @@ class Picklist(TimeStampedModel):
     class Meta:
         abstract = True
 
+    def __str__(self):
+        return '{}-{}'.format(self.value, self.label)
+
 
 class TicketStatus(Picklist):
     pass
 
     class Meta:
-        verbose_name = 'Ticket status'
         verbose_name_plural = 'Ticket statuses'
 
+
+class TicketPriority(Picklist):
+    pass
+
+    class Meta:
+        verbose_name_plural = 'Ticket priorities'
+
+
+class Queue(Picklist):
+    pass
+
+
+class ProjectStatus(Picklist):
+    pass
+
+    class Meta:
+        verbose_name_plural = 'Project statuses'
+
+
+class ProjectType(Picklist):
+    pass
+
+
+class Resource(TimeStampedModel):
+    user_name = models.CharField(max_length=32)
+    email = models.CharField(max_length=50)
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    active = models.BooleanField(default=False)
+
+    class Meta:
+        ordering = ('first_name', 'last_name')
+
     def __str__(self):
-        return '{}-{}'.format(self.value, self.label)
+        return '{} {}'.format(self.first_name, self.last_name)
+
+
+class TicketSecondaryResource(TimeStampedModel):
+    resource = models.ForeignKey(
+        'Resource', blank=True, null=True, on_delete=models.SET_NULL
+    )
+    ticket = models.ForeignKey(
+        'Ticket', blank=True, null=True, on_delete=models.SET_NULL
+    )
+
+    def __str__(self):
+        return '{} {}'.format(self.resource, self.ticket)
+
+
+class Account(TimeStampedModel):
+    name = models.CharField(max_length=100)
+    number = models.CharField(max_length=50)
+    active = models.BooleanField(default=True)
+    last_activity_date = models.DateTimeField(blank=True, null=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Project(TimeStampedModel):
+    name = models.CharField(max_length=100)
+    number = models.CharField(null=True, max_length=50)
+    description = models.CharField(max_length=2000)
+    actual_hours = models.DecimalField(
+        null=True, decimal_places=2, max_digits=9)
+    completed_date_time = models.DateTimeField(null=True)
+    completed_percentage = models.PositiveSmallIntegerField(default=0)
+    duration = models.PositiveSmallIntegerField(default=0)
+    start_date_time = models.DateTimeField(null=True)
```

Comment:
Huh. They really want to track their projects closely I guess. "THE PROJECT WILL START FRIDAY AT EXACTLY 2:54 PM! AND NOT A MINUTE SOONER!"

Reply:
Actually no, its not. I checked autotask and it looks like they display it as a date only and ignore the time even though it comes in as a date time. We can probably store it as a date field too then just like we do in djconnectwise. Unless there is a good reason not to. I feel like ambiguous project start/end dates based on time zones is not good though.

Reply:
Good point. Might as well take the same approach as django-connectwise. :+1: